File: 2do_DAM/PMDM/1ra_Evaluacion/Examen/ExamenPrueba01/Examen01_Eduardo_Martin-Sonseca/app/src/main/java/com/example/examen01_eduardo_martin_sonseca/MainActivity.java
package com.example.examen01_eduardo_martin_sonseca;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.ListView;
import android.widget.ScrollView;
import android.widget.Spinner;
import android.widget.Toast;
import android.widget.ToggleButton;

import com.example.examen01_eduardo_martin_sonseca.databinding.ActivityMainBinding;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    private ArrayList<String> almacenaDatos = new ArrayList<>();


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // setContentView(R.layout.activity_main);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        String nombre = binding.etNombre.getText().toString();
        String edad = binding.etEdad.getText().toString();
        String ciudades = (String) binding.spCiudades.getSelectedItem();
        ToggleButton activarDesactivar = binding.tbOnOff;
        Button bGrabar = binding.bGrabar;
        ListView lvDatos = binding.lvDatos;

        activarDesactivar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (activarDesactivar.getText().equals("OFF")){
                    activarDesactivar.setText("ON");
                    bGrabar.setEnabled(true);
                } else{
                    activarDesactivar.setText("OFF");
                    bGrabar.setEnabled(false);
                }
            }
        });

        bGrabar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String nombre = binding.etNombre.getText().toString();
                String edad = binding.etEdad.getText().toString();
                String ciudades = (String) binding.spCiudades.getSelectedItem();

                Persona persona = new Persona(nombre,edad,ciudades);



                // Actualizar el ListView
                ArrayAdapter<String> adaptado = new ArrayAdapter<>(getApplicationContext(), android.R.layout.simple_list_item_1, almacenaDatos);
                lvDatos.setAdapter(adaptado);
                almacenaDatos.add(persona.toString());
                adaptado.notifyDataSetChanged();
            }
        });

        lvDatos.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {

                Persona persona = new Persona(nombre,edad,ciudades);
                Toast.makeText(getApplicationContext(), "Los datos son " + persona.toString(), Toast.LENGTH_LONG).show();

            }
        });
    }


    }

File: 2do_DAM/PMDM/1ra_Evaluacion/Ejercicios/Kotlin/Ejercicio09_Menus_Dialogo/app/src/main/java/com/example/ejercicio09_menus_dialogo/MainActivity.kt
package com.example.ejercicio09_menus_dialogo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import com.example.ejercicio09_menus_dialogo.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    lateinit var binding : ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // setContentView(R.layout.activity_main)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        var btnFinalizar = binding.bFinalizar

        btnFinalizar.setOnClickListener(View.OnClickListener {
            var alerta = AlertDialog.Builder(this)
            alerta.setTitle("Confirmación")
            alerta.setMessage("¿Confirma que deseas salir?")
            alerta.setPositiveButton("Si"){dialog, which ->
                Toast.makeText(this, "Has pulsado el boton Ok",Toast.LENGTH_LONG).show()
            }
            alerta.setNegativeButton("No"){dialog, which->
                Toast.makeText(this, "Has pulsado el boton No",Toast.LENGTH_LONG).show()
            }
            alerta.show()
        })
    }


}